lib/database-ops.js

```diff
@@ -137,9 +137,12 @@ var startDatabaseClient = function(callback, options) {
             }
             else {
                 imageEntryCollection = database.collection("image-entries");
-                imageEntryCollection.dropIndex("expire_index");
-                createIndexForImageEntriesCollection(imageEntryCollection)
-                finishInitialization();
+                imageEntryCollection.dropIndex("expire_index").then(() => {
+                    createIndexForImageEntriesCollection(imageEntryCollection)
+                    finishInitialization();
+                }).catch((err) => {
+                    callback(err);
```

Comment:
In the case that `dropIndex` fails due to index not existing, then we should create the index and not call the callback with error. You can check `err.codeName` and if it's "IndexNotFound" then you can call create index method and finish initialization.